src/features/AnnouncementModal/AnnouncementModal.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Flex,
+  Image,
+  Modal,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  ModalBody,
+  Stack,
+  Link,
+} from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+import { ProgressIndicator } from "components/ProgressIndicator"
+import { useCallback, useMemo, useState } from "react"
+import { BiRightArrowAlt } from "react-icons/bi"
+
+import { useAnnouncements } from "hooks/useAnnouncement"
+
+import { Announcement } from "types/announcements"
+
+import { NewFeatureTag } from "./components/NewFeatureTag"
+
+interface AnnouncementModalProps {
+  isOpen: boolean
+  onClose: () => void
+  announcements: Announcement[]
+  link: string
+}
+
+export const AnnouncementModal = ({
+  isOpen,
+  announcements,
+  onClose,
+  link,
+}: AnnouncementModalProps): JSX.Element => {
+  const [currActiveIdx, setCurrActiveIdx] = useState<number>(0)
+  const { setLastSeenAnnouncement } = useAnnouncements()
+  const isLastAnnouncement = useMemo(
+    () => currActiveIdx === announcements.length - 1,
+    [announcements.length, currActiveIdx]
+  )
+
+  const handleNextClick = useCallback(() => {
```

Comment:
Why do we need useCallback here?

Reply:
so that we avoid excessive re-rendering if any of the deps change.

Reply:
is there value add cos of useCallback here though. is a simple handleNextClick.

If there are deep child component dependencies this would help

Reply:
Sure, I'll remove it!